classes/log/EmailLogEntry.php

```diff
@@ -11,230 +11,143 @@
  * @ingroup log
  *
- * @see EmailLogDAO
- *
  * @brief Describes an entry in the email log.
  */
 
 namespace PKP\log;
 
+use Illuminate\Database\Eloquent\Casts\Attribute;
+use Illuminate\Database\Eloquent\Model;
 use APP\facades\Repo;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Support\Str;
+use Eloquence\Behaviours\HasCamelCasing;
+use PKP\log\core\EmailLogEventType;
 
-class EmailLogEntry extends \PKP\core\DataObject
+class EmailLogEntry extends Model
 {
-    //
-    // Get/set methods
-    //
-
-    /**
-     * Get user ID of sender.
-     *
-     * @return int
-     */
-    public function getSenderId()
-    {
-        return $this->getData('senderId');
-    }
-
-    /**
-     * Set user ID of sender.
-     *
-     * @param int $senderId
-     */
-    public function setSenderId($senderId)
-    {
-        $this->setData('senderId', $senderId);
-    }
-
-    /**
-     * Get date email was sent.
-     *
-     * @return string
-     */
-    public function getDateSent()
-    {
-        return $this->getData('dateSent');
-    }
-
-    /**
-     * Set date email was sent.
-     *
-     * @param string $dateSent
-     */
-    public function setDateSent($dateSent)
-    {
-        $this->setData('dateSent', $dateSent);
-    }
-
-    /**
-     * Get event type.
-     *
-     * @return int
-     */
-    public function getEventType()
-    {
-        return $this->getData('eventType');
-    }
+    use HasCamelCasing;
+
+    protected $table = 'email_log';
+    protected $primaryKey = 'log_id';
+    public $timestamps = false;
+
+    protected $fillable = [
+        'assocType',
+        'assocId',
+        'senderId',
+        'dateSent',
+        'eventType',
+        'from',
+        'recipients',
+        'subject',
+        'body',
+        'bccRecipients',
+        'ccRecipients',
+        'fromAddress'
+    ];
 
     /**
-     * Set event type.
+     * The maximum length for the email subject.
      *
-     * @param int $eventType
+     * This value should match the length of the `subject` column in the `email_log` table, defined in LogMigration.php.
      */
-    public function setEventType($eventType)
-    {
-        $this->setData('eventType', $eventType);
-    }
+    private const MAX_SUBJECT_LENGTH = 255;
 
-    /**
-     * Get associated type.
-     *
-     * @return int
-     */
-    public function getAssocType()
+    protected static function booted(): void
     {
-        return $this->getData('assocType');
+        static::creating(function (EmailLogEntry $entry) {
+            $subject = $entry->subject;
+            // Subtract 3 to compensate for the '...' that gets added to the end of the string.
+            $entry->subject = Str::limit($subject, self::MAX_SUBJECT_LENGTH - 3);
+        });
     }
 
-    /**
-     * Set associated type.
-     *
-     * @param int $assocType
-     */
-    public function setAssocType($assocType)
+    //
+    // Accessors / Mutators
+    //
+    protected function id(): Attribute
```

Comment:
I kept this accessor/mutator for the primary key `log_id` because `$email->id` seems more standard and intuitive than `$email->logId`.